common/src/main/java/com/tc/net/protocol/transport/ClientMessageTransport.java

```diff
@@ -112,30 +115,72 @@ public NetworkStackID open() throws TCTimeoutException, IOException, MaxConnecti
     this.isOpening.set(true);
     synchronized (this.isOpen) {
       Assert.eval("can't open an already open transport", !this.isOpen.get());
-      this.connectionEstablisher.open(this);
-      Assert.eval(!this.connectionId.isNull());
+      Assert.eval("can't open an already connected transport", !this.isConnected());
+
+      TCSocketAddress socket = new TCSocketAddress(info);
+      TCConnection connection = connect(socket);
+      try {
+        openConnection(connection);
+      } catch (CommStackMismatchException e) {
+        connection.close(100);
+        throw e;
+      } catch (MaxConnectionsExceededException e) {
+        connection.close(100);
+        throw e;
+      } catch (TCTimeoutException e) {
+        connection.close(100);
+        throw e;
+      } catch (TransportHandshakeException e) {
+        connection.close(100);
+        throw e;
+      }
+      Assert.eval(!getConnectionId().isNull());
       this.isOpen.set(true);
-      NetworkStackID nid = new NetworkStackID(this.connectionId.getChannelID());
+      NetworkStackID nid = new NetworkStackID(getConnectionId().getChannelID());
       this.wasOpened = true;
       this.isOpening.set(false);
       return (nid);
     }
   }
-
+  /**
+   * Tries to make a connection. This is a blocking call.
+   * 
+   * @return
+   * @throws TCTimeoutException
+   * @throws IOException
+   * @throws MaxConnectionsExceededException
+   */
+  TCConnection connect(TCSocketAddress sa) throws TCTimeoutException, IOException {
+    TCConnection connection = this.connectionManager.createConnection(getProtocolAdapter());
+    fireTransportConnectAttemptEvent();
+    try {
+      connection.connect(sa, this.timeout);
+    } catch (IOException e) {
+      connection.close(100);
+      throw e;
+    } catch (TCTimeoutException e) {
+      throw e;
```

Comment:
It looks like TCTimeout causes the underlying connection to do its own cleanup but adding the close for safe measure seems like a good idea.